### PR TITLE
feat!: replace function calls with map lookups

### PR DESCRIPTION
**Pull request by arty-name**
## fixes KILTProtocol/ticket#2040

The functions seem to bring no value, and even their names are longer

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: packages/did/src/Did.utils.ts [hunk 1]
```diff
@@ -124,72 +124,34 @@ export function isSameSubject(didA: DidUri, didB: DidUri): boolean {
   return parseDidUri(didA).address === parseDidUri(didB).address
 }
 
-const signatureAlgForKeyType: Record<VerificationKeyType, SigningAlgorithms> = {
+export const signatureAlgForKeyType: Record<
+  VerificationKeyType,
+  SigningAlgorithms
+> = {
```
Comment (rflechtner) [inline packages/did/src/Did.utils.ts:130]:
```suggestion
export const signatureAlgForKeyType: Record<
  VerificationKeyType,
  SigningAlgorithms
> = Object.freeze({
```

File: packages/did/src/Did.utils.ts [hunk 2]
```diff
@@ -124,72 +124,34 @@ export function isSameSubject(didA: DidUri, didB: DidUri): boolean {
...
-  return keyTypeForSignatureAlg[signatureAlg]
-}
-
-const encryptionAlgForKeyType: Record<EncryptionKeyType, EncryptionAlgorithms> =
-  {
-    x25519: 'x25519-xsalsa20-poly1305',
-  }
+export const keyTypeForSignatureAlg = Object.entries(
+  signatureAlgForKeyType
+).reduce((obj, [key, value]) => ({ ...obj, [value]: key }), {}) as Record<
```
Comment (rflechtner) [inline packages/did/src/Did.utils.ts:138]:
```suggestion
export const keyTypeForSignatureAlg = Object.freeze(
  Object.entries(
  signatureAlgForKeyType
).reduce((obj, [key, value]) => ({ ...obj, [value]: key }), {})) as Record<
```

File: packages/did/src/Did.utils.ts [hunk 3]
```diff
@@ -124,72 +124,34 @@ export function isSameSubject(didA: DidUri, didB: DidUri): boolean {
...
- * Given the identifier of a key type, returns the identifier of the encryption algorithm for which it is used.
- *
- * @param keyType Key type identifier.
- * @returns Encryption algorithm identifier.
- */
-export function getEncryptionAlgorithmForEncryptionKeyType(
-  keyType: EncryptionKeyType
-): EncryptionAlgorithms {
-  return encryptionAlgForKeyType[keyType]
+export const encryptionAlgForKeyType: Record<
```
Comment (rflechtner) [inline packages/did/src/Did.utils.ts:143]: same

File: packages/did/src/Did.utils.ts [hunk 4]
```diff
@@ -124,72 +124,34 @@ export function isSameSubject(didA: DidUri, didB: DidUri): boolean {
...
- * Given the identifier of an encryption algorithm, returns the identifier of the key type required for it.
- *
- * @param encryptionAlg Encryption algorithm identifier.
- * @returns Key type identifier.
- */
-export function getEncryptionKeyTypeForEncryptionAlgorithm(
-  encryptionAlg: EncryptionAlgorithms
-): EncryptionKeyType {
-  return keyTypeForEncryptionAlg[encryptionAlg]
+export const keyTypeForEncryptionAlg: Record<
```
Comment (rflechtner) [inline packages/did/src/Did.utils.ts:150]: same

**Review by rflechtner — COMMENTED** [review]
Exporting the maps technically opens up the possibility to mess with these mappings, which I think was one of the reasons for the function approach. But how about we just add an `Object.freeze` here?

**Review by rflechtner — APPROVED** [review]
LGTM!

**Comment by arty-name** [comment]
I could live with `Object.freeze`, but the dynamic nature of JS still allows anyone to mess with the modules to the great extent. Whenever the SDK code accesses these maps as `Did.Utils.encryptionAlgForKeyType`, the external developer can simply do `Did.Utils.encryptionAlgForKeyType = { foo: bar }`. With these glaring holes being open by the dynamic nature of the language itself, I don’t see much use in attempts to protect something.

**Pull request merged**